### PR TITLE
Bug 1357946 - Time waits for no tab

### PR DESCRIPTION
**Pull request by rnewman**
On the face of it, this code was safe. But that cast to `Timestamp` — which is the same as `UInt64` — would fail for negative numbers or very large numbers. This PR tightens that up a lot, handles our own mis-generated millisecond timestamps, handles doubles in records, and correctly produces decimal seconds.

### CODE REVIEW

**Comment by farhanpatel** [comment]
### SwiftLint found issues

#### Errors

File | Line | Reason |
| --- | ----- | ----- |
TimeConstants.swift | 84 | Lines should not have trailing semicolons. 
TimeConstants.swift | 114 | Lines should not have trailing semicolons. 
TimeConstants.swift | 139 | Lines should not have trailing semicolons. 
UtilsTests.swift | 67 | Lines should not have trailing semicolons. 
UtilsTests.swift | 68 | Lines should not have trailing semicolons. 
UtilsTests.swift | 69 | Lines should not have trailing semicolons. 
UtilsTests.swift | 71 | Lines should not have trailing semicolons. 
UtilsTests.swift | 72 | Lines should not have trailing semicolons. 
UtilsTests.swift | 73 | Lines should not have trailing semicolons. 
UtilsTests.swift | 75 | Lines should not have trailing semicolons. 
UtilsTests.swift | 76 | Lines should not have trailing semicolons. 
UtilsTests.swift | 77 | Lines should not have trailing semicolons. 
UtilsTests.swift | 79 | Lines should not have trailing semicolons. 
UtilsTests.swift | 80 | Lines should not have trailing semicolons. 
UtilsTests.swift | 82 | Lines should not have trailing semicolons. 
UtilsTests.swift | 83 | Lines should not have trailing semicolons. 
UtilsTests.swift | 86 | Lines should not have trailing semicolons. 
TabsPayload.swift | 12 | Lines should not have trailing semicolons. 
TabsPayload.swift | 13 | Lines should not have trailing semicolons. 


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: Sync/TabsPayload.swift [hunk 1]
```diff
@@ -8,6 +8,10 @@ import Storage
 import XCGLogger
 import SwiftyJSON
 
+// Int64.max / 1000.
+private let MaxSecondsToConvertInt64: Int64 = 9223372036854775;
+private let MaxSecondsToConvertDouble: Double = Double(9223372036854775);
```
Comment (justindarc) [inline Sync/TabsPayload.swift]: nit: Can't we just set these equal to `Int64.max / 1000` and `Double(Int64.max / 1000)` respectively?

File: Client.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -4093,7 +4096,7 @@
 				TargetAttributes = {
 					2827315D1ABC9BE600AA1954 = {
 						CreatedOnToolsVersion = 6.2;
-						DevelopmentTeam = 9G8J6YA743;
+						DevelopmentTeam = 43AQ936H96;
```
Comment (justindarc) [inline Client.xcodeproj/project.pbxproj]: @st3fan did you mean to change the DevelopmentTeam?

File: Sync/TabsPayload.swift (hunk 1 above)
Comment (justindarc) [inline Sync/TabsPayload.swift]: Actually... nevermind.. for some reason `Double(Int64.max / 1000)` evaluates to `9223372036854776` in Swift playground.

Comment (justindarc) [inline Sync/TabsPayload.swift]: Err... I thought it was a rounding thing, but it turns out that `Double(9223372036854775)` hard-coded also evaluates to `9223372036854776` in a Swift playground

Comment (justindarc) [inline Sync/TabsPayload.swift]: D'oh.. just realized that's a Swift Playground anomaly... Its actually setting it properly

**Pull request merged**